vpr/src/route/rr_graph.cpp

```diff
@@ -1469,40 +1470,60 @@ static void build_rr_sinks_sources(const int i,
                         } else {
                             VTR_ASSERT(class_inf[iclass].type == DRIVER);
                             //Initialize the output pin
                             // Note that we leave it's out-going edges unconnected (they will be hooked up to global routing later)
                             inode = get_rr_node_index(L_rr_node_indices, i + width_offset, j + height_offset, OPIN, ipin, side);
 
-                            //Initially left unconnected
-                            VTR_ASSERT(inode >= 0);
-                            L_rr_node[inode].set_cost_index(OPIN_COST_INDEX);
-                            L_rr_node[inode].set_type(OPIN);
+                            /* Output pins may not exist on some sides, we may not always find one 
+                             * And to avoid duplicated edges, we just create edge the first time 
+                             * we find the node
+                             */
+                            if ((OPEN != inode) && (true == first_pin)) {
+                                //Initially left unconnected
+                                VTR_ASSERT(inode >= 0);
+                                L_rr_node[inode].set_cost_index(OPIN_COST_INDEX);
+                                L_rr_node[inode].set_type(OPIN);
+                            }
                         }
 
                         /* Common to both DRIVERs and RECEIVERs */
-                        L_rr_node[inode].set_capacity(1);
-                        float R = 0.;
-                        float C = 0.;
-                        L_rr_node[inode].set_rc_index(find_create_rr_rc_data(R, C));
-                        L_rr_node[inode].set_ptc_num(ipin);
-
-                        //Note that we store the grid tile location and side where the pin is located,
-                        //which greatly simplifies the drawing code
-                        L_rr_node[inode].set_coordinates(i + width_offset, j + height_offset, i + width_offset, j + height_offset);
-                        L_rr_node[inode].set_side(side);
-
-                        VTR_ASSERT(type->pinloc[width_offset][height_offset][L_rr_node[inode].side()][L_rr_node[inode].pin_num()]);
+                        if ((OPEN != inode) && (true == first_pin)) {
+                            L_rr_node[inode].set_capacity(1);
+                            float R = 0.;
+                            float C = 0.;
+                            L_rr_node[inode].set_rc_index(find_create_rr_rc_data(R, C));
+                            L_rr_node[inode].set_ptc_num(ipin);
+
+                            //Note that we store the grid tile location and side where the pin is located,
+                            //which greatly simplifies the drawing code
+                            //For those pins located on multiple sides, we only allocate one rr_node and
+                            //store the grid tile location and side for the first pin we find
```

Comment:
Is this comment on "only store the grid tile location and side for the first pin we find" correct?  I think we should be saving the rr-node index for the pin on all sides and offsets at which it exists.